### PR TITLE
feat(app): Render update app modal if update available

### LINKED ISSUE #1154: App Update: Display Dialogue Box on Page View
As a user, I would like to be notified that a robot update is available as soon as I view the Robot Settings page.

## Acceptance Criteria
- If an app update is available, display the 'Update Available' dialogue box as soon as the user views the App Settings page.

## Design
- Can use the same dialogue box that we show when a user clicks the 'Update' button:
- https://projects.invisionapp.com/d/main#/console/13154070/275374445/preview
- ^ this is the robot update dialogue box design, but you can just swap 'server' for 'app'

### PR DESCRIPTION
**Pull request by Kadee80**
## overview
This PR closes #1154 by rendering the app update modal on app page load if 
- an update is available
- the update alert has not been marked seen

screen shot 2018-05-30 at 4 57 42 pm

## changelog

- add `SET_UPDATE_SEEN` reducer to shell
- set `update: {... seen: true}` on `START_DOWNLOAD`
- redirect `/menu/app` to `menu/app/update` on app page view if update is available and seen=false
- update `START_DOWNLOAD` test and write `SET_UPDATE_SEEN` in shell
- add alertOverlay prop `AppUpdateModal`

## review requests

Please let me know if tests are missing.

To test UI (no robot neccessary):

1. Change `app-shell/package.json` version (line 4) from `"version": "3.0.1-beta.1"` to "version": "3.0.0"
2. Click 'more' nav button in main nav
- [x] app setting page renders app update modal on initial page view
- [x] in redux devtools shell.update.seen is false

3.  Click [Not now] button in Alert
- [x] modal closes
- [x] returning to app from another page (same session) does not render the modal again
- [x] in redux devtools shell.update.seen is now true

### CODE REVIEW

File: app/src/pages/AppSettings.js [hunk 1]
```diff
@@ -60,7 +72,10 @@ function mapDispatchToProps (dispatch: Dispatch): DP {
     checkForUpdates: () => dispatch(checkForShellUpdates()),
     downloadUpdate: () => dispatch(downloadShellUpdate()),
     quitAndInstall: () => quitAndInstallShellUpdate(),
-    closeUpdateModal: () => dispatch(push('/menu/app')),
+    closeUpdateModal: () => {
+      dispatch(push('/menu/app'))
+      dispatch(setUpdateSeen())
```
Comment (mcous) [inline app/src/pages/AppSettings.js]: Can we switch the order of these to be safe? It's not happening on my app, but I'm a little worried about a little redirect race-condition if the `push` happens before `update.seen` is set to true, resulting in another redirect

**Comment by mcous** [comment]
Can you also do me a solid and sneak in a fix for [a bug in the close button text](https://github.com/Opentrons/opentrons/pull/1559#discussion_r191931943) I introduced in #1559?

In AppUpdateModal.js, the top of the function should be:

```js
export default function AppUpdateModal (props: Props) {
  const {close, update: {error, downloadInProgress}} = props
  const {button, message} = mapPropsToButtonPropsAndMessage(props)
  const closeButtonChildren = error || downloadInProgress
    ? 'close'
    : 'not now'
```

File: app/src/pages/AppSettings.js (hunk 1 above)
Comment (Kadee80) [reply app/src/pages/AppSettings.js]: 👍

File: app/src/components/AppSettings/AppUpdateModal.js [hunk 2]
```diff
@@ -16,9 +16,9 @@ type Props = {
 const Spinner = () => (<Icon name='ot-spinner' height='1em' spin />)
 
 export default function AppUpdateModal (props: Props) {
-  const {close, update: {downloadInProgress}} = props
+  const {close, update: {error, downloadInProgress}} = props
```
Comment (mcous) [inline app/src/components/AppSettings/AppUpdateModal.js]: 🙏

**Comment by IanLondon** [comment]
Talked with you all: this doesn't work on Linux b/c there's no linux yaml on S3. Mike will open issue.

:penguin: :disappointed: 

We should probably test this on Windows too while we're at it, right?

**Comment by mcous** [comment]
Works on my install of Windows 10

**Pull request merged**